src/main/java/Duke.java

```diff
@@ -1,10 +1,414 @@
+            if (nextLine.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            }
+            parser.parse();
+        }
+    }
+
+    public static void main (String[]args) throws DukeException {
+        new Duke("listStore.ser").run();
+    }
+
+    // Input class represents user inputted tasks
+    public static class Input implements Serializable {
+        boolean done;
+        String content;
+        String id;
+        LocalDate time;
+        String printTime;
+
+        Input(String content) {
+            this.content = content;
+            boolean done = false;
+        }
+
+        Input(boolean done, String content) {
+            this.done = done;
+            this.content = content;
+        }
+
+        // Marks task as done
+        public void taskDone() {
+            this.done = true;
+        }
+    }
+
+    public static class Todo extends Input {
+
+        Todo(String content) {
+            super(content);
+            this.id = "[T]";
+            this.time = null;
+            this.printTime = "";
+        }
+
+        Todo(boolean done, String content) {
+            super(done, content);
+            this.id = "[T]";
+            this.time = null;
+            this.printTime = "";
+        }
+    }
+
+    public static class Deadline extends Input {
+
+        // Constructor for Deadline Object, converts time string to LocalDate and formats for printing
+        Deadline(String content, String deadlineTime) {
+            super(content);
+            this.time = LocalDate.parse(deadlineTime, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            this.id = "[D]";
+            this.printTime = "(" + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+
+        Deadline(boolean done, String content, String deadlineTime) {
+            super(done, content);
+            this.time = LocalDate.parse(deadlineTime, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            this.id = "[D]";
+            this.printTime = "(" + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+    }
+
+    public static class Event extends Input {
+
+        // Constructor for Event Object, converts time string to LocalDate and formats for printing
+        Event(String content, String eventTime) {
+            super(content);
+            this.time = LocalDate.parse(eventTime, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            this.id = "[E]";
+            this.printTime = "(" + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+
+        Event(boolean done, String content, String eventTime) {
+            super(done, content);
+            this.time = LocalDate.parse(eventTime, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            this.id = "[E]";
+            this.printTime = "(" + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+    }
```

Comment:
I am glad that all of your code is maintained below the line length limit which is below 120 char, so the code view is better. 👍